### PR TITLE
Incorporate vuejs

### PR DESCRIPTION
**Pull request by dsamojlenko**
Incorporates [VueJS](https://vuejs.org/v2/guide/) to the build pipeline.

Taking a light approach where we can "sprinkle in" bits of dynamic components where needed using [Single File Components](https://vuejs.org/v2/guide/single-file-components.html). May in future look at enabling routes to return a full Vue template.

The way it works:

- Create a vue component in the `routes/[route]/js/` folder
- Add vue bootstrapping  and component registration code to the .js file in that folder
- Use the custom component in the route view, ie: `<example></example>`
- Profit!

To test this out, there is a an example component included in the `personal` route. Open the personal.njk template, add `<example></example>` somewhere on the page, wait for it to rebuild, and viola!

Note that Single File Component CSS blocks are currently not working - probably just need to configure a loader. I think this is ok, since we've included TailwindCSS, we shouldn't need to write custom styles. We can look at fixing if we find we need this.

Also, this approach doesn't lend itself well to reusing components in other routes - we can change approach later if needed.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -66,6 +66,13 @@ app.use(compression())
 app.locals.GITHUB_SHA = process.env.GITHUB_SHA || null
 app.locals.hasData = hasData
 
+// Using the getClientJs to grab the global app.js and set it in app.locals so we can render in templates
+const { getClientJs } = require('./utils')
+app.use(function(req, res, next) {
+  app.locals.appJs = getClientJs(req, 'app')
+  next()
+})
+
```
Comment (dsamojlenko) [inline app.js]: I'm not crazy about this bit. Open to alternative suggestions.

The issue is, webpack is compiling all of our js/css assets using chunkhash filenames. For page-specific js, we have a helper that checks the manifest (`_filelist.json`) and pulls in relevant js files for the route. But what to do with a global app.js?

Ultimately, it works, but there might be a better way.

**Comment by CalvinRodo** [comment]
Did you update a bunch of dependencies in this PR as well as adding Vue?

File: package.json [hunk 2]
```diff
@@ -31,6 +31,9 @@
   },
   "dependencies": {
     "@cdssnc/webpack-starter": "^2.0.2",
+    "@csstools/postcss-sass": "^4.0.0",
+    "@fullhuman/postcss-purgecss": "^1.3.0",
+    "axios": "^0.19.0",
```
Comment (CalvinRodo) [inline package.json]: Did you add axios?

I'm not seeing it used in the PR, and I'm not seeing the previous version in package.json

File: app.js (hunk 1 above)
Comment (CalvinRodo) [reply app.js]: Do we have to add vue app wide? Can we just keep it at page specific for now?

File: package.json [hunk 3]
```diff
@@ -45,18 +48,19 @@
     "express-validator": "^6.2.0",
     "helmet": "^3.21.2",
     "i18n": "^0.8.4",
+    "knex": "^0.20.1",
```
Comment (jeffmaher) [inline package.json]: Similar to Calvin's earlier comment, looks like maybe some things slipped in that are unrelated to this PR. Probably want to remove this in case the Sequalize PR gets merged first.

File: package.json [hunk 4]
```diff
@@ -45,18 +48,19 @@
...
     "mini-css-extract-plugin": "^0.8.0",
     "nunjucks": "^3.2.0",
+    "pg": "^7.12.1",
```
Comment (jeffmaher) [inline package.json]: Unrelated dependency?

**Review by jeffmaher — CHANGES_REQUESTED** [review]
Tested in Chrome and looks good. Left a couple comments about dependency leakage for adjustment.

File: package.json (hunk 3 above)
Comment (CalvinRodo) [reply package.json]: Knex got merged in through an earlier PR

File: package.json (hunk 4 above)
Comment (CalvinRodo) [reply package.json]: This was merged in in an earlier PR, I think it's just a version update, it is being used in my PR so no biggie for me.

File: package.json (hunk 2 above)
Comment (dsamojlenko) [reply package.json]: was from a previous iteration of the code ... removed.

File: app.js (hunk 1 above)
Comment (dsamojlenko) [reply app.js]: see latest

File: package.json (hunk 4 above)
Comment (dsamojlenko) [reply package.json]: I think most of these package.json changes were just a re-ordering - when using `npm install`, npm organizes depedencies alphabetically, so if you've previously added deps the end of the list, they can get reorganized by another `npm install` command.

**Review by CalvinRodo — APPROVED** [review]
🥇

**Review by jeffmaher — APPROVED** [review]
